numpy/core/src/multiarray/mapping.c

```diff
@@ -169,7 +169,7 @@ prepare_index(PyArrayObject *self, PyObject *index,
     int new_ndim, fancy_ndim, used_ndim, index_ndim;
     int curr_idx, get_idx;
 
-    npy_intp i, n;
+    int i, n;
```

Comment:
This doesn't look right as `PySequence_Size` is later assigned to `n`
```
Py_ssize_t PySequence_Size(PyObject *o)
```
`npy_intp` is effectively the same as `Py_ssize_t`. Note that one needs to be *very* careful in replaceing `npy_intp` by `int`.